### PR TITLE
Add unit tests for FIRCLSRecord*

### PR DESCRIPTION
**Pull request by jasonhu-g**
Add unit tests to validate the logic of reading the .clsrecord files

### CODE REVIEW

File: Crashlytics/UnitTests/FIRCLSRecordTests.m [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+  // Verify first
+  XCTAssertTrue([adapter.keyValues[0].key isEqualToString:@"com.crashlytics.in-background"]);
+  XCTAssertTrue([adapter.keyValues[0].value isEqualToString:@"0"]);
+
+  // Verify last
+  XCTAssertTrue([adapter.keyValues[11].key isEqualToString:@"com.crashlytics.user-id"]);
+  XCTAssertTrue([adapter.keyValues[11].value
+      isEqualToString:@"test-user-28AE6E09-BC30-4CB3-9FA5-FE06828B8F3C"]);
+}
```
Comment (samedson) [inline Crashlytics/UnitTests/FIRCLSRecordTests.m]: What do you think about storing a key-value line with an emoji in it, and then reading it out here to make sure they work?

File: Crashlytics/UnitTests/FIRCLSRecordTests.m [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+@interface FIRCLSRecordTests : XCTestCase
+
+@end
+
+@implementation FIRCLSRecordTests
+
+/// It is important that crashes do not occur when reading persisted crash files before uploading
+/// Verify various invalid input cases
+- (void)testInvalidRecordCases {
+  id adapter __unused = [[FIRCLSRecordAdapter alloc] initWithPath:@"nonExistentPath"];
```
Comment (samedson) [inline Crashlytics/UnitTests/FIRCLSRecordTests.m]: Could we add a test case where we initialize with a valid path, but files in the path are corrupted / invalid JSON?

File: Crashlytics/UnitTests/FIRCLSRecordTests.m (hunk 1 above)
Comment (jasonhu-g) [reply Crashlytics/UnitTests/FIRCLSRecordTests.m]: I remove this test case as I am now using FIRCLSFile's decode/encode.

This test case would work:
- (void)testHexDecoding {
  NSString *str = @"Hello world! 👻";
  NSString *encoded =  FIRCLSFileHexEncodeString([str UTF8String]);
  NSString *decoded =  FIRCLSFileHexDecodeString([encoded UTF8String]);
  XCTAssertTrue([str isEqualToString:decoded]);
}

File: Crashlytics/UnitTests/FIRCLSRecordTests.m (hunk 2 above)
Comment (jasonhu-g) [reply Crashlytics/UnitTests/FIRCLSRecordTests.m]: Added.

**Pull request merged**